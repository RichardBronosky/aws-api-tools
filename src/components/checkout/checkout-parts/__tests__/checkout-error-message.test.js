import React from 'react'
import renderer from 'react-test-renderer'
import CheckoutErrorMessage from '../checkout-error-message'

describe('CheckoutErrorMessage', () => {
  const testProps = {
    invalidFields: [],
    customMessage: null,
  }

  it('does not render without invalid fields or custom message', () => {
    const tree = renderer.create(<CheckoutErrorMessage { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with 1 invalid field', () => {
    const newTestProps = {
      ...testProps,
      invalidFields: ['phone'],
    }
    const tree = renderer.create(<CheckoutErrorMessage { ...newTestProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with 2 invalid fields', () => {
    const newTestProps = {
      ...testProps,
      invalidFields: ['first name', 'phone'],
    }
    const tree = renderer.create(<CheckoutErrorMessage { ...newTestProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with 3 invalid fields', () => {
    const newTestProps = {
      ...testProps,
      invalidFields: ['first name', 'last name', 'phone'],
    }
    const tree = renderer.create(<CheckoutErrorMessage { ...newTestProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with 4 invalid fields', () => {
    const newTestProps = {
      ...testProps,
      invalidFields: ['first name', 'last name', 'phone', 'zip'],
    }
    const tree = renderer.create(<CheckoutErrorMessage { ...newTestProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with custom error message and no invalid fields', () => {
    const newTestProps = {
      ...testProps,
      customMessage: 'Test error message',
    }
    const tree = renderer.create(<CheckoutErrorMessage { ...newTestProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with custom error message and invalid fields (should not show invalid fields)', () => {
    const newTestProps = {
      invalidFields: ['first name', 'last name', 'phone', 'zip'],
      customMessage: 'Test error message',
    }
    const tree = renderer.create(<CheckoutErrorMessage { ...newTestProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
