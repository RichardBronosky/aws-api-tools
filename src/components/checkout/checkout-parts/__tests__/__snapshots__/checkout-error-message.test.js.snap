// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutErrorMessage does not render without invalid fields or custom message 1`] = `null`;

exports[`CheckoutErrorMessage renders correctly with 1 invalid field 1`] = `
<p
  className="invalid-message"
>
  *Invalid
   
  phone
</p>
`;

exports[`CheckoutErrorMessage renders correctly with 2 invalid fields 1`] = `
<p
  className="invalid-message"
>
  *Invalid
   
  first name 
  and phone
</p>
`;

exports[`CheckoutErrorMessage renders correctly with 3 invalid fields 1`] = `
<p
  className="invalid-message"
>
  *Invalid
   
  first name, 
  last name, 
  and phone
</p>
`;

exports[`CheckoutErrorMessage renders correctly with 4 invalid fields 1`] = `
<p
  className="invalid-message"
>
  *Invalid
   
  first name, 
  last name, 
  phone, 
  and zip
</p>
`;

exports[`CheckoutErrorMessage renders correctly with custom error message and invalid fields (should not show invalid fields) 1`] = `
<p
  className="invalid-message"
>
  *
  Test error message
</p>
`;

exports[`CheckoutErrorMessage renders correctly with custom error message and no invalid fields 1`] = `
<p
  className="invalid-message"
>
  *
  Test error message
</p>
`;
